### ISSUE TITLE
Error in nbconvert with inline matplotlib plots

### ISSUE THREAD

**Issue by crmccreary** [jupyter/nbconvert#930]
The cell data type of an inline matplotlib figure is bytes, not  str.
So nbconvert/preprocessors/extractoutput.py attempts `data = json.dumps(data)` which results in the error

`TypeError: Object of type 'bytes' is not JSON serializable`

My fix is

```
84c84
<                         not isinstance(data, text_type)
---
>                         not (isinstance(data, text_type) or isinstance(data, bytes))

```
but this is not backwards compatible to Python 2.7.

[test_nbconvert.zip](https://github.com/jupyter/nbconvert/files/2690439/test_nbconvert.zip)

**Comment by crmccreary** [comment]
Perhaps a better fix which is backwards compatible with 2.7 is

`20c20
<     text_type = str
---
>     text_type = (str, bytes)`

**Comment by t-makaro** [comment]
What exporter are you using? I believe that this is fixed in nbconvert 5.5.0 by #985.

**Comment by Plinsboorg** [comment]
Hi @crmccreary 
Please, can you check whether your issue is solved in the nbconvert 5.5.0+ ?

**Comment by crmccreary** [comment]
Yes
